src/rp2_common/pico_aon_timer/aon_timer.c

```diff
@@ -56,6 +58,10 @@ bool aon_timer_set_time_calendar(const struct tm *tm) {
     datetime_t dt;
     tm_to_datetime(tm, &dt);
     rtc_set_datetime(&dt);
+
+    // Writing to the RTC will take 2 clk_rtc clock periods to arrive
+    uint rtc_freq = clock_get_hz(clk_rtc);
+    busy_wait_us((1000000 / rtc_freq) * 2);
```

Comment:
I presume you probably want `(((1000000 + rtc_freq - 1) / rtc_freq) * 2)` for rounding safety